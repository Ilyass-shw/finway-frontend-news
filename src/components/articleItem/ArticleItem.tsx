import React from "react";
import { Image, Text, View } from "react-native";
import { article } from "../../types";
import { styles } from "./ArticleItemStyles";

interface ArticleItemProps {
  article: article;
}
const ArticleItem: React.FC<ArticleItemProps> = ({ article }) => {
  return (
    <View style={styles.itemWrapperStyle}>
      <Image
        style={styles.itemImageStyle}
        source={{ uri: article.urlToImage }}
      />
      <View style={styles.contentWrapperStyle}>
        <View style={styles.TitleContainer}>
          <Text
            numberOfLines={3}
            ellipsizeMode="tail"
            style={styles.txtTitleStyle}
          >
            {article.title}
          </Text>
        </View>
        <View>
          <Text style={styles.sourceName}>{article.source.name}</Text>
        </View>
      </View>
    </View>
  );
};

export default ArticleItem;
