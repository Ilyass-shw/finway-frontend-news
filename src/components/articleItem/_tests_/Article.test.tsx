import { render, fireEvent } from "@testing-library/react-native";
import ArticleItem from "../ArticleItem";

test("ArticleItem should render", () => {

    const article = {
      source: {
        id: "engadget",
        name: "Engadget",
      },
      author: "Igor Bonifacic",
      title:
        "The latest 'Belle' trailer introduces us to the film's dazzling technicolor virtual world",
      description:
        "With Mamoru Hosoda’s latest movie opening in US theaters this Friday, Studio Chizu and the film’s distributor have shared a new trailer for Belle\r\n. The more than three-minute-long clip shows the movie’s opening scene in its entirety, introducing us to U, Bel…",
      url: "https://www.engadget.com/belle-trailer-u-virtual-world-234024067.html",
      urlToImage:
        "https://s.yimg.com/os/creatr-uploaded-images/2022-01/c8d44190-7334-11ec-bd7d-84d991cbfb48",
      publishedAt: "2022-01-11T23:40:24Z",
      content:
        "With Mamoru Hosodas latest movie opening in US theaters this Friday, Studio Chizu and the films distributor have shared a new trailer for Belle\r\n. The more than three-minute-long clip shows the movie… [+781 chars]",
    };

  const { toJSON } = render(
    <ArticleItem article={article}/>
  );

  expect(toJSON()).toMatchSnapshot();
});
