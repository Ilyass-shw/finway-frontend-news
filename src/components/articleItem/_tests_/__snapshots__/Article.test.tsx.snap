// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleItem should render 1`] = `
<View
  style={
    Object {
      "alignSelf": "center",
      "backgroundColor": "white",
      "flexDirection": "row",
      "height": 160,
      "marginVertical": 4,
      "width": "90%",
    }
  }
>
  <Image
    source={
      Object {
        "uri": "https://s.yimg.com/os/creatr-uploaded-images/2022-01/c8d44190-7334-11ec-bd7d-84d991cbfb48",
      }
    }
    style={
      Object {
        "marginHorizontal": 9,
        "marginVertical": 12,
        "width": "50%",
      }
    }
  />
  <View
    style={
      Object {
        "width": "80%",
      }
    }
  >
    <View
      style={
        Object {
          "height": "60%",
          "marginTop": 14,
          "width": "53%",
        }
      }
    >
      <Text
        ellipsizeMode="tail"
        numberOfLines={3}
        style={
          Object {
            "fontSize": 17,
            "fontWeight": "700",
          }
        }
      >
        The latest 'Belle' trailer introduces us to the film's dazzling technicolor virtual world
      </Text>
    </View>
    <View>
      <Text
        style={
          Object {
            "fontSize": 14,
            "marginTop": 8,
          }
        }
      >
        Engadget
      </Text>
    </View>
  </View>
</View>
`;
